functions/imagemagick/ImageMagick/Function.cs

```diff
@@ -0,0 +1,171 @@
+[assembly: FunctionsStartup(typeof(ImageMagick.Startup))]
+
+namespace ImageMagick
+{
+    // Dependency injection configuration, executed during server startup.
+    public class Startup : FunctionsStartup
+    {
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            builder.Services
+                .AddSingleton(ImageAnnotatorClient.Create())
+                .AddSingleton(StorageClient.Create());
+        }
+    }
+
+    /// <summary>
+    /// A function that can be triggered in responses to changes in Google Cloud Storage.
+    /// The type argument (StorageObjectData in this case) determines how the event payload is deserialized.
+    /// The event must be deployed so that the trigger matches the expected payload type. (For example,
+    /// deploying a function expecting a StorageObject payload will not work for a trigger that provides
+    /// a FirestoreEvent.)
+    /// </summary>
+    public class Function : ICloudEventFunction<StorageObjectData>
+    {
+        /// <summary>
+        /// The bucket to store blurred images in. An alternative to using environment variables here would be to
+        /// fetch it from IConfiguration.
+        /// </summary>
+        private static readonly string s_blurredBucketName = Environment.GetEnvironmentVariable("BLURRED_BUCKET_NAME");
+
+        private readonly ImageAnnotatorClient _visionClient;
+        private readonly StorageClient _storageClient;
+        private readonly ILogger _logger;
+
+        public Function(ImageAnnotatorClient visionClient, StorageClient storageClient, ILogger<Function> logger) =>
+            (_visionClient, _storageClient, _logger) = (visionClient, storageClient, logger);
+
+        // [END functions_imagemagick_setup]
+
+        // [START functions_imagemagick_analyze]
+        public async Task HandleAsync(CloudEvent cloudEvent, StorageObjectData data, CancellationToken cancellationToken)
+        {
+            // Validate parameters
+            if (data.Bucket is null || data.Name is null)
+            {
+                _logger.LogError("Malformed GCS event.");
+                return;
+            }
+
+            // Construct URI to GCS bucket and file.
+            string gcsUri = $"gs://{data.Bucket}/{data.Name}";
+            _logger.LogInformation("Analyzing {uri}", gcsUri);
+
+            // Perform safe search detection using the Vision API.
+            Image image = Image.FromUri(gcsUri);
+            SafeSearchAnnotation annotation;
+            try
+            {
+                annotation = await _visionClient.DetectSafeSearchAsync(image);
+            }
+            // If the call to the Vision API fails, log the error but let the function complete normally.
+            // If the exceptions weren't caught (and just propagated) the event would be retried.
+            // See the "Best Practices" section in the documentation for more details about retry.
```

Comment:
nit: Is there a link that's stable?

Reply:
I don't know how stable any such link would be, to be honest. I'd prefer to stay a bit vague for now.